statistics.rb

```diff
@@ -0,0 +1,33 @@
+module Statistics
+  # def top(orders, object, quantity = 1)
+  #   tops = Hash.new(0)
+  #   orders.each { |order| tops[order.send(object)] += 1 }
+  #   tops.sort_by { |_object, number| -number }[0...quantity].to_h
+  # end
+  #
+  # Why this code doesn't working as it should???
+  # def top(orders, object, quantity = 1)
+  #   tops = Hash.new(0)
+  #   orders.group_by(&object).each { |order| tops[order[0]] += 1}
+  #   tops.sort_by { |_object, number| -number }[0...quantity].to_h
+  # end
+
+  def top(orders, object, quantity = 1)
+    tops = Hash.new(0)
+
+    case object
+    when :book
+      orders.each { |order| tops[order.book] += 1 }
+    when :reader
+      orders.each { |order| tops[order.reader] += 1 }
+    end
+
+    tops.sort_by { |_object, number| -number }[0...quantity].to_h
+  end
```

Comment:
Could you please avoid populating empty hash during iteration through orders?
This approach is twice slowest that approach with using `group_by`

```
require 'benchmark'
orders = 1_000_000.times.map { Order.new(books.sample, readers.sample, FFaker::Time.date) }

def top_with_group_by(orders, object, attr_name, quantity = 1)
  orders.map(&object)
        .group_by(&attr_name)
        .sort_by { |grouped, entitites| -entitites.size }
        .first(quantity)
        .map(&:first)
end

#  0.753150   0.000779   0.753929 (  0.754288)
Benchmark.measure { top(orders, :book, 1) }

#  0.339748   0.000000   0.339748 (  0.340723)
Benchmark.measure { top_with_group_by(orders, :book, :title, 1) }
```